IBPSA/Fluid/HeatPumps/ModularReversible/Modular.mo

```diff
@@ -0,0 +1,136 @@
+    con(preDro(m_flow(nominal=QHea_flow_nominal/1000/10))),
+    eva(preDro(m_flow(nominal=QHea_flow_nominal/1000/10))),
+    final PEle_nominal=refCyc.refCycHeaPumHea.PEle_nominal,
+    mCon_flow_nominal=QHea_flow_nominal/(dTCon_nominal*cpCon),
+    mEva_flow_nominal=(QHea_flow_nominal - PEle_nominal)/(dTEva_nominal*cpEva),
+    use_rev=true,
+    redeclare IBPSA.Fluid.HeatPumps.ModularReversible.BaseClasses.RefrigerantCycle refCyc(
+      redeclare model RefrigerantCycleHeatPumpHeating =
+          RefrigerantCycleHeatPumpHeating,
+      redeclare model RefrigerantCycleHeatPumpCooling =
+          RefrigerantCycleHeatPumpCooling));
+  parameter Modelica.Units.SI.HeatFlowRate QHea_flow_nominal(min=Modelica.Constants.eps)
+    "Nominal heating capacity"
+    annotation (Dialog(group="Nominal condition"));
+  parameter Modelica.Units.SI.HeatFlowRate QCoo_flow_nominal(max=0)=0
+    "Nominal cooling capacity"
+      annotation(Dialog(group="Nominal condition - Cooling", enable=use_rev));
+
+  replaceable model RefrigerantCycleHeatPumpHeating =
+    IBPSA.Fluid.HeatPumps.ModularReversible.RefrigerantCycle.BaseClasses.PartialHeatPumpCycle
+      (PEle_nominal=0)
+       constrainedby
+    IBPSA.Fluid.HeatPumps.ModularReversible.RefrigerantCycle.BaseClasses.PartialHeatPumpCycle(
+       final useInHeaPum=true,
+       final QHea_flow_nominal=QHea_flow_nominal,
+       final TCon_nominal=TConHea_nominal,
+       final TEva_nominal=TEvaHea_nominal,
+       final cpCon=cpCon,
+       final cpEva=cpEva)
+  "Refrigerant cycle module for the heating mode"
+    annotation (choicesAllMatching=true);
+
+  replaceable model RefrigerantCycleHeatPumpCooling =
+      IBPSA.Fluid.Chillers.ModularReversible.RefrigerantCycle.BaseClasses.NoCooling
+      constrainedby
+    IBPSA.Fluid.Chillers.ModularReversible.RefrigerantCycle.BaseClasses.PartialChillerCycle(
+       final useInChi=false,
+       final cpCon=cpCon,
+       final cpEva=cpEva,
+       final TCon_nominal=TEvaCoo_nominal,
+       final TEva_nominal=TConCoo_nominal,
+       QCoo_flow_nominal=QCoo_flow_nominal)
+  "Refrigerant cycle module for the cooling mode"
+    annotation (Dialog(enable=use_rev),choicesAllMatching=true);
+  parameter Modelica.Units.SI.Temperature TConHea_nominal
+    "Nominal temperature at secondary condenser side in heating mode"
+    annotation (Dialog(group="Nominal condition"));
+  parameter Modelica.Units.SI.Temperature TEvaHea_nominal
+    "Nominal temperature at secondary evaporator side in heating mode"
+    annotation (Dialog(group="Nominal condition"));
+  parameter Modelica.Units.SI.Temperature TConCoo_nominal
+    "Nominal temperature at secondary condenser side in cooling mode"
+    annotation(Dialog(enable=use_rev, group="Nominal condition - Cooling"));
+  parameter Modelica.Units.SI.Temperature TEvaCoo_nominal
+    "Nominal temperature at secondary evaporator side in cooling mode"
+    annotation(Dialog(enable=use_rev, group="Nominal condition - Cooling"));
```

Comment:
Yes, you got that right. It depends on how you look at it. The condenser in the sense of condensing the fluid, yes. But the condenser in the sense of the phyiscal component, then no. The same would apply for every other variable, e.g. `QCon_flow` which is, in cooling mode, also the evaporator in the thermodynamic sense. 

We've been over this discussion multiple times, there is no 100 % best way to name these sides. I can change the description, but the name I am afraid will stay.

Reply:
I understand that the variable names would be hard to change now.
I still think that the description strings should be changed though. My experience feedback is that, after examining the models in detail, I was still unable to provide the expected parameters here and initially used mixed up design temperatures. What about: `Load side fluid temperature` and `Source side fluid temperature`?
(Mentioning `in cooling mode` is redundant as the information is already provided in the parameter group name.
Unless you think it avoids ambiguities in the variable explorer when analyzing simulation results – as the description string can also be looked up here.)
Same changes needed for the heating mode parameters.

Reply:
What do you think about the following descriptions to avoid ambiguities?
```
  parameter Modelica.Units.SI.Temperature TConCoo_nominal
    "Nominal temperature at secondary ambient side (condenser)"
    annotation(Dialog(enable=use_rev, group="Nominal condition"));
  parameter Modelica.Units.SI.Temperature TEvaCoo_nominal
    "Nominal temperature at secondary load side (evaporator)"
    annotation(Dialog(enable=use_rev, group="Nominal condition"));
  parameter Modelica.Units.SI.Temperature TConHea_nominal=TConCoo_nominal
    "Nominal temperature at secondary ambient side (condenser of cooling mode)"
    annotation (Dialog(group="Nominal condition - Heating"));
  parameter Modelica.Units.SI.Temperature TEvaHea_nominal=TEvaCoo_nominal
    "Nominal temperature at secondary load side (evaporator of cooling mode)"
    annotation (Dialog(group="Nominal condition - Heating"));
```

Reply:
- I think the ambiguities remain as long as you keep the words evaporator and condenser in the description strings, even within parentheses.
- I think the word "source" is more appropriate than "ambient" because HP can be connected to a condenser water loop, exhaust air, district loop, etc. and "ambient" does not reflect this. Also, the terms "air-source" and "water-source" heat pumps are widely used.
- When using "source-side fluid" and "load-side fluid", the term "secondary" is not needed anymore.
- I don't see in which configurations the default bindings would be valid: the design HW temperature will almost never match the design outdoor air temperature (or brine temperature) in cooling mode and the design CHW temperature will almost never match the design outdoor air temperature (or brine temperature) in heating mode. I think no default can be provided here.

Reply:
I get that the ambiguities remain, but the variable itself is named Eva and Con. Not explaining why in the doc is also an ambiguity in my opinion. Source may be more appropriate for heat pumps, but "source" is again not practical for chillers, or? Because users could think that the chiller source is the fluid being cooled down. I still like source and load and will add it to the UserGuide and the doc of these variables. 

What do you think about the following?
```
  parameter Modelica.Units.SI.Temperature TConCoo_nominal
    "Nominal temperature at source side, the condenser"
    annotation(Dialog(enable=use_rev, group="Nominal condition"));
  parameter Modelica.Units.SI.Temperature TEvaCoo_nominal
    "Nominal temperature at load side, the evaporator"
    annotation(Dialog(enable=use_rev, group="Nominal condition"));
  parameter Modelica.Units.SI.Temperature TConHea_nominal
    "Nominal temperature at source side, the condenser of the cooling mode"
    annotation (Dialog(group="Nominal condition - Heating"));
  parameter Modelica.Units.SI.Temperature TEvaHea_nominal
    "Nominal temperature at load side, the evaporator of the cooling mode"
    annotation (Dialog(group="Nominal condition - Heating"));
```

Reply:
Sorry, I was focused on HP and forgot about chiller applications. "Source" is indeed less appropriate for chillers, but I don't think there is any ambiguity and that a user could misinterpret the source side as the CHW loop. 
ASHRAE Guideline 36 also provides the following definitions and examples:
> A component is a “source” if it provides resources to a downstream component
> A component is a “load” if it receives resources from an upstream component
> The cooling towers together constitute a system, which is a source to the chillers.
> The CHW pumps together constitute a system, which is a load to the chillers and a source to the air handlers.

Sorry to insist but for me, "condenser" and "evaporator" are too misleading in the case of reversible systems. 
Maybe having other developers' feedback may be useful?

Is it intentional to have deleted the mention of the cooling mode in the group name of `TConCoo_nominal` and `TEvaCoo_nominal`? It must appear somewhere, right?

I think native English speakers would say "on the source/load side" rather than "at source/load side".

Reply:
@AntoineGautier had I great idea about solving this issue on the call we just had, just call it `Nominal temperature of the heated fluid` (or `cooled`) should always remove ambiguities. I will update the doc and UserGuide accordingly.